lib/active_record/connection_adapters/postgis_adapter/rails3/spatial_column_info.rb

```diff
@@ -0,0 +1,42 @@
+module ActiveRecord  # :nodoc:
+  module ConnectionAdapters  # :nodoc:
+    module PostGISAdapter
+      # Do spatial sql queries for column info and memoize that info.
+      class SpatialColumnInfo
+        def initialize(adapter, table_name)
+          @adapter = adapter
+          @table_name = table_name
+        end
+
+        def all
+          info = @adapter.query("SELECT f_geometry_column,coord_dimension,srid,type FROM geometry_columns WHERE f_table_name='#{@table_name}'")
+          result = {}
+          info.each do |row|
+            name = row[0]
+            type = row[3]
+            dimension = row[1].to_i
+            has_m = !!(type =~ /m$/i)
+            type.sub!(/m$/, '')
+            has_z = dimension > 3 || dimension == 3 && !has_m
```

Comment:
I also found this line difficult to read.  I thought to myself "Why not `dimension >= 3 && !has_m` ?  And I had to google the subtlety that && has higher precedence than || which means it's really trying to say `(dimension > 3 /* and don't care about has_m */) || (dimension == 3 && !has_m)`

Again, if you're porting from code we didn't write, don't change a thing because it'll help future ports to keep it the same.  But if we're writing this from scratch, I'd prefer parentheses, at least in the latter part.